ci(lint): various improvements

lewis6991 opened the pull request:
- Added https://github.com/reviewdog/action-suggester for uncrustify results:
  - The workflow doesn't have the required level of permissions to post suggestion comments, so the step fallbacks to writing annotated messages in the log.
- Run `lintc` instead of `lintcfull`:
   - `lintcfull` == `lintc` + `uncrustify`, however `uncrustify` already has it's own step.
- Separated `lintc` from `lint`:
  - Before `lint` took around `3m30s`
  - Now `lint` takes around `1m30s` and `lintc` takes around `3m`, BUT, they are done in parallel.

lewis6991 committed:
ci(lint): generate PR suggestions from uncrustify

lewis6991 committed:
ci(lint): run uncrustify earlier

clason commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,152 @@
+        name: lintcfull
+        run: make lintcfull
+
+      - if: "!cancelled()"
```
can we separate these out into an earlier job so we don't have to build just to notice stylua failures?

clason commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,152 @@
+        name: lintcfull
+        run: make lintcfull
+
+      - if: "!cancelled()"
+        name: lintstylua
+        uses: JohnnyMorganz/stylua-action@1.0.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --check runtime/
+
+      - if: "!cancelled()"
+        name: lintlua
+        run: make lintlua
+
+      - if: "!cancelled()"
+        name: lintpy
+        run: make lintpy
+
+      - if: "!cancelled()"
+        name: lintsh
+        run: make lintsh
```
can we separate these out into an earlier job so we don't have to build just to notice stylua failures?

clason commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,152 @@
+        run: ./ci/run_tests.sh build_nvim
+
+      - if: "!cancelled()"
+        name: lintcfull
```
do we need to run the full `lintcfull` again if we already did `uncrustify` before? Might just run `lintc` here.

lewis6991 committed:
ci(line): lintcfull -> lintc

lintcfull == lintc + uncrustify

We are already running uncrustify separately so can just run lintc
instead.

lewis6991 committed:
ci(lint): move some steps before build

lewis6991 committed:
ci(lint): separate lintc from lint

All steps in lint now do now require a Neovim build.
